xarray/tests/test_backends.py

```diff
@@ -512,6 +512,7 @@ def test_roundtrip_string_data(self) -> None:
         with self.roundtrip(expected) as actual:
             assert_identical(expected, actual)
 
+    @pytest.mark.filterwarnings("ignore:Setting encoding directly.*:FutureWarning")
```

Comment:
I've explicitly marked all tests that use the `encoding` setter with this filter. When we remove the setter, we'll need to come back through and rework these tests.